pkg/yurttunnel/server/interceptor.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"bufio"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/util/httpstream"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/klog"
+)
+
+// RequRequestInterceptor intercepts http/https requests sent from the master,
+// prometheus and metric server, setup proxy tunnel to kubelet, sends requests
+// through the tunnel and sends responses back to the master
+type RequestInterceptor struct {
+	UDSSockFile string
+	TLSConfig   *tls.Config
+}
```

Comment:
would you add requestWrapper slice for RequestInterceptor in order to modify request for user customization?